electrumx/server/env.py

```diff
@@ -43,6 +43,9 @@ def __init__(self, coin=None):
 
         # Core items
 
+        self.is_public = self.boolean('PUBLIC', False)
+        self.is_watchtower = self.boolean('WATCHTOWER', False)
+        assert not (self.is_public and self.is_watchtower)
```

Comment:
```suggestion
        if self.is_public and self.is_watchtower:
            raise cls.Error("WATCHTOWERs are not safe to use atm for PUBLIC servers")
```
---
note: operators of public servers might want to use the watchtower functionality for themselves though. So we could allow PUBLIC and WATCHTOWER, but restrict the watchtower-related RPCs to the whitelist.